### PR TITLE
Port To Python 3

### PR DESCRIPTION
**Pull request by Creatune**
Please let me know if the changes are valid.

### CODE REVIEW

File: mcpi/minecraft.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def __init__(self, connection, packagePrefix):
     def getPos(self, id):
         """Get entity position (entityId:int) => Vec3"""
         s = self.conn.sendReceive(self.pkg + ".getPos", id)
-        return Vec3(*map(float, s.split(",")))
+        return Vec3(*list(map(float, s.split(","))))
```
Comment (dbrgn) [inline mcpi/minecraft.py]: Are you sure this is required? It seems that list unpacking works with iterators as well.

```python
>>> def make(a, b, c):
...     return (a, b, c)
...
>>> make(*map(float, '1,2,3'.split(',')))
(1.0, 2.0, 3.0)
```

File: mcpi/mock_server.py [hunk 2]
```diff
@@ -21,4 +21,4 @@ class ThreadedServer(socketserver.ThreadingMixIn, socketserver.TCPServer):
     server_thread = threading.Thread(target=server.serve_forever)
     server_thread.daemon = False
     server_thread.start()
-    print("server is now running on port {}".format(PORT))
+    print(("server is now running on port {}".format(PORT)))
```
Comment (dbrgn) [inline mcpi/mock_server.py]: These parentheses are redundant.

File: mcpi/util.py [hunk 3]
```diff
@@ -1,19 +1,19 @@
...
+    for e in l:
+        if isinstance(e, collections.Iterable) and not isinstance(e, basestring):
+            for ee in flatten(e):
+                yield ee
+        else:
+            yield e
+
+
+def flatten_parameters_to_string(l):
+    return ",".join(map(str, flatten(l)))
```
Comment (dbrgn) [inline mcpi/util.py:19]: Why is the diff so large? Did you convert line endings?

**Review by dbrgn — CHANGES_REQUESTED** [review]
I'm a bit confused. This PR does not seem to do any Python 3 porting at all.

File: mcpi/util.py [hunk 4]
```diff
@@ -1,14 +1,14 @@
 import collections
 
 try:
-    str
+    basestring
 except NameError:
-    str = (str, bytes)
+    basestring = (str, bytes)
```
Comment (dbrgn) [inline mcpi/util.py]: Basestring is a Python 2 only construct. If anything, basestring should be removed, not added.

**Comment by Creatune** [comment]
I used the 2to3 tool to convert. So all the changes (except for changing the str to basestring) was done by that tool.

**Comment by dbrgn** [comment]
> I used the 2to3 tool to convert. So all the changes (except for changing the str to basestring) was done by that tool.

Ah. 2to3 often does a bit too much, since it's not very sophisticated.

In any case, py3minepi already supports Python 3, so it doesn't require any porting :slightly_smiling_face: Python 3 support is the whole reason why this repo exists.

But even then, the official Minecraft Python support is Python 3 compatible by now: https://projects.raspberrypi.org/en/projects/getting-started-with-minecraft-pi @py3minepi/admins maybe the repos could be archived?

**Pull request closed without merging**